integration-test/click-to-load-facebook.spec.js

```diff
@@ -0,0 +1,226 @@
+import { test, expect } from './helpers/playwrightHarness'
+import backgroundWait from './helpers/backgroundWait'
+import { loadTestConfig, loadTestTds } from './helpers/testConfig'
+import { routeFromLocalhost } from './helpers/testPages'
+import { getDomain } from 'tldts'
+
+const testSite = 'https://privacy-test-pages.glitch.me/privacy-protections/click-to-load/'
+const facebookDomains = new Set(['facebook.com', 'facebook.net', 'fbcdn.net'])
+
+/**
+ * Start logging requests for the given Puppeteer Page.
+ * @param {import('@playwright/test').Page} page
+ *   The Puppeteer page to log requests for.
+ * @param {LoggedRequestDetails[]} requests
+ *   Array of request details, appended to as requests happen.
+ *   Note: The requests array is mutated by this function.
+ * @param {function} [filter]
+ *   Optional filter function that (if given) should return falsey for requests
+ *   that should be ignored.
+ * @returns {Promise<function>}
+ *   Function that clears logged requests (and in progress requests).
+ */
+async function logPageRequests (page, requests, filter) {
```

Comment:
Perhaps put this in `helpers/requests.js`? In fact, this looks very similar to the existing helper there, so perhaps you could expand the existing helper to support Playwright?